EnvironmentSimulator/Modules/RoadManager/RoadManager.cpp

```diff
@@ -1418,6 +1422,19 @@ void Road::AddElevation(Elevation *elevation)
 	elevation_profile_.push_back((Elevation*)elevation);
 }
 
+void Road::AddSuperElevation(Elevation *super_elevation)
+{
+	// Adjust last super elevation length
+	if (super_elevation_profile_.size() > 0)
+	{
+		Elevation *e_previous = super_elevation_profile_.back();
+		e_previous->SetLength(super_elevation->GetS() - e_previous->GetS());
+	}
+	super_elevation->SetLength(length_ - super_elevation->GetS());
```

Comment:
Ah, smart, each added superelevation entry gets length = remaining length (s) of the road. Which is then adjusted when/if another is added. Nice.